lib/semantics/resolve-names.cc

```diff
@@ -2483,6 +2483,13 @@ void SubprogramVisitor::Post(const parser::FunctionStmt &stmt) {
   funcResultDetails.set_funcResult(true);
   funcInfo_.resultSymbol =
       &MakeSymbol(*funcResultName, std::move(funcResultDetails));
+  if (funcInfo_.resultName && funcInfo_.resultName->source == name.source) {
+    // C1560. TODO also enforce on entry names when entry implemented
+    Say(funcInfo_.resultName->source,
+        "'%s' is already the function name and cannot appear in RESULT"_err_en_US,
```

Comment:
I would emit a warning, not an error, and ignore the `RESULT` name.

Reply:
OK, pgfortran is accepting this (none of the other tested compilers do) and I do not see issues that could come from relaxing the constraint.
I have transformed the error into a warning that states that references to `f` inside the function scope will be considered references to the result only.